dashboard/app/lib/javascripts/dashboard/client.js

```diff
@@ -97,6 +97,25 @@ Dashboard.Client = Marbles.Utils.createClass({
 		});
 	},
 
+	ping: function (protocol) {
+		var controllerKey = (Dashboard.config.user || {}).controller_key,
+			endpoint;
+
+		console.log(controllerKey, protocol, this.endpoints, this);
+
+		protocol = protocol === undefined ? 'https' : protocol;
+		endpoint = this.endpoints.cluster_controller.substr(0, 5) === "https" ?
+			protocol + endpoint.substr(5) :
+			protocol + endpoint.substr(4);
```

Comment:
[yes](https://github.com/flynn/flynn/blob/master/dashboard/api.go#L182), so no need even for the regex (`replace("https", protocol)` will work).